### PR TITLE
Kinesis + Kafka connector docs

### PR DESCRIPTION
**Pull request by oliviamiannone**
**Description:**

Add docs for Kinesis and Kafka connectors

**Workflow steps:**

NA

**Documentation links affected:**

NA

**Notes for reviewers:**
As always, this is a technical review to ensure I captured the appropriate level of detail for users and didn't misinterpret anything along the way. It seemed reasonable to combine two connector docs into one PR; I'm sure it's obvious but I'm only asking you to review the connector you wrote (:

A few specific requests
- (BOTH) - For both these connectors, I'm unsure what the role of `namespace` is in the binding config, or if it needs to be reflected here. Please advise!
- (KAFKA) - as noted in the code sample, I could use some good example auth parameters

---
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/380)

### CODE REVIEW

File: site/docs/reference/Connectors/capture-connectors/apache-kafka.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  * Connection security enabled with TLS (highly recommended for production environments)
+
+### Authentication and connection security
+
+Neither authentication nor connection security are enabled by default in your Kafka cluster, but both are important considerations.
+Similarly, Flow's Kafka connectors do not strictly require authentication or connection security mechanisms.
+You may choose to omit them for local development and testing; however, both are strongly encouraged for production environments.
+
+A wide [variety of authentication methods](https://kafka.apache.org/documentation/#security_overview) is available in Kafka clusters.
+All SASL methods are permissable for clusters used with Flow, so long as connection details are provided.
```
Comment (saterus) [inline site/docs/reference/Connectors/capture-connectors/apache-kafka.md:24]: I'm not 100% sure that we support all of them. I've setup/tested SCRAM-256, SCRAM-512, and PLAIN. I vaguely remember the other SASL mechanisms being more complicated than just username+password.

File: site/docs/reference/Connectors/capture-connectors/apache-kafka.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+### Sample
+```yaml
+captures:
+  ${TENANT}/${CAPTURE_NAME}:
+    endpoint:
+      connector:
+        image: ghcr.io/estuary/source-kafka:dev
+        config:
+            bootstrap_servers: localhost:9093
```
Comment (saterus) [inline site/docs/reference/Connectors/capture-connectors/apache-kafka.md:64]: I think this needs to be a list of servers.

```suggestion
            bootstrap_servers: [localhost:9093]
```

File: site/docs/reference/Connectors/capture-connectors/apache-kafka.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            tls: system_certificates
+            authentication: #FOR REVIEW: need semi-realistic example values
+                mechanism:
+                username:
+                password:
```
Comment (saterus) [inline site/docs/reference/Connectors/capture-connectors/apache-kafka.md:69]: The only restriction here would be on the `mechanism` field, and it's one of `PLAIN`, `SCRAM-SHA-256`, or `SCRAM-SHA-512`.

```suggestion
            authentication:
                mechanism: SCRAM-SHA-512
                username: bruce.wayne
                password: definitely-not-batman
```

File: site/docs/reference/Connectors/capture-connectors/apache-kafka.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    bindings:
+      - resource:
+           namespace: ${STREAM_NAMESPACE} #FOR REVIEW: does namespace matter?/What does it do here?
```
Comment (saterus) [inline site/docs/reference/Connectors/capture-connectors/apache-kafka.md:72]: I think `namespace` is an optional Airbyte-spec thing. I don't believe we use it for anything.

File: site/docs/reference/Connectors/capture-connectors/apache-kafka.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+When authentication details are not provided, the client connection will attempt to use PLAINTEXT (insecure) protocol.
+
+If you don't already have authentication enabled on your cluster, Estuary recommends [SASL/SCRAM](https://kafka.apache.org/documentation/#security_sasl_scram).
+With SCRAM, you set up a username and password, making it analogous to the traditional authentication mechanisms
+you use in other applications.
+
+For connection security, Estuary recommends that you enable TLS encryption for your SASL mechanism of choice,
+as well as all other components of your cluster.
+Note that because TLS replaced now-deprecated SSL encryption, Kafka still uses the acronym "SSL" to refer to TLS encryption.
+See [Confluent's documentation](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html) for details.
```
Comment (saterus) [inline site/docs/reference/Connectors/capture-connectors/apache-kafka.md:34]: Good note. :+1:

**Review by saterus — APPROVED** [review]
This looks great. Thanks for writing all this up! :1st_place_medal:

File: site/docs/reference/Connectors/capture-connectors/apache-kafka.md (hunk 3 above)
Comment (oliviamiannone) [reply site/docs/reference/Connectors/capture-connectors/apache-kafka.md:69]: Thanks. I am using this exact example.

**Comment by oliviamiannone** [comment]
> This looks good to me. I did realize that we didn't document the specific AWS permissions that are needed for the user. I forgot that in the README, too. I think it's fine to merge this and then circle back to that, though, since it'll probably take a little research on my part to verify the exact permissions we need.

Leaving the docs pending tag on this PR so we don't lose track. For now, a note has been added to the Kinesis docs.

**Pull request merged**